File: aigc-cartoon-react/src/api/storyboard.ts
import { get, post, put, del } from './request';

export interface Storyboard {
  id?: number;
  scriptId: number;
  sceneNumber: number;
  sceneDescription?: string;
  duration?: number;
  bgPrompt?: string;
  characterPrompt?: string;
  audioUrl?: string;
  createTime?: string;
  updateTime?: string;
}

export interface StoryboardForm {
  scriptId: number;
  sceneNumber: number;
  sceneDescription?: string;
  duration?: number;
  bgPrompt?: string;
  characterPrompt?: string;
  audioUrl?: string;
}

export const getStoryboards = (scriptId: number) =>
  get<Storyboard[]>(`/scripts/${scriptId}/storyboards`);

export const getStoryboardDetail = (id: number) =>
  get<Storyboard>(`/storyboards/${id}`);

export const createStoryboard = (data: StoryboardForm) =>
  post<Storyboard>('/storyboards', data);

export const updateStoryboard = (id: number, data: Partial<Storyboard>) =>
  put<boolean>(`/storyboards/${id}`, data);

export const deleteStoryboard = (id: number) =>
  del<boolean>(`/storyboards/${id}`);

export const batchCreateStoryboard = (storyboards: StoryboardForm[]) =>
  post<boolean>('/storyboards/batch', storyboards);

File: aigc-cartoon-server/src/main/java/com/aigc/cartoon/entity/Storyboard.java
package com.aigc.cartoon.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_storyboard")
public class Storyboard {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long scriptId;
    
    private Integer sceneNumber;
    
    private String sceneDescription;
    
    private Integer duration;
    
    private String bgPrompt;
    
    private String characterPrompt;
    
    private String audioUrl;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    
    @TableLogic
    private Integer deleted;
}